### PR TITLE
Introducing a pypdf CLI program - close #2160

### LINKED ISSUE #2160: Feature suggestion: provide pdfrw examples as commands
## Explanation

`pdfrw` provides a set of very useful scripts: https://github.com/sarnold/pdfrw/tree/master/examples
Personnally, I very often use:
* [cat.py](https://github.com/sarnold/pdfrw/blob/master/examples/cat.py) to concatenate/merge multiple PDFs
* [subset.py](https://github.com/sarnold/pdfrw/blob/master/examples/subset.py) to extract some pages from a PDF

I think that a script to extract all images from a PDF would also be handy, like this one: https://stackoverflow.com/a/34116472/636849

The idea would be to expose those features as simple CLI `console_scripts`:
https://github.com/Lucas-C/pre-commit-hooks/blob/master/setup.py#L27

This way, a user could simply:
```
pip install pypdf
pypdf_subset in.pdf 31-34 [out.pdf]
pypdf_cat in.pdf out.pdf merged.pdf
pypdf_extract_images merged.pdf
```

Would you be open to welcome such feature in `py-pdf`?

### PR DESCRIPTION
**Pull request by Lucas-C**
What's missing:
* maybe more tests?
* a dedicated documentation page?

### CODE REVIEW

File: pypdf/filters.py [hunk 1]
```diff
@@ -847,7 +847,7 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
         lookup: Any
         base: Any
         hival: Any
-        if isinstance(color_space, ArrayObject) and color_space[0] == "/Indexed":
+        if isinstance(color_space, ArrayObject) and len(color_space) == 4 and color_space[0] == "/Indexed":
```
Comment (stefan6419846) [inline pypdf/filters.py:850]: How is this change related to the CLI?

Comment (Lucas-C) [reply pypdf/filters.py:850]: This is a fixup for a bug.

Introducing this CLI and the related unit tests revealed this issue:
```
  File "/opt/pypdf/pypdf/filters.py", line 851, in _handle_flate
    color_space, base, hival, lookup = (
ValueError: not enough values to unpack (expected 4, got 3)
```

There is also another bug spotted there:
https://github.com/py-pdf/pypdf/blob/issue-2160/tests/test_cli.py#L91

**Comment by Lucas-C** [comment]
This PR can probably be closed, given that https://github.com/py-pdf/pdfly offers similar functionalities

File: pypdf/filters.py (hunk 1 above)
Comment (pubpub-zz) [reply pypdf/filters.py:850]: @Lucas-C 
can you provide the pdf showing the issue

Comment (Lucas-C) [reply pypdf/filters.py:850]: Yes, this is one of `py-pdf` test samples: https://github.com/py-pdf/pypdf/blob/main/resources/AutoCad_Diagram.pdf

The error was raised by simply calling `pypdf extract-images AutoCad_Diagram.pdf`

Comment (Lucas-C) [reply pypdf/filters.py:850]: I reported the 2nd bug there: https://github.com/py-pdf/pypdf/issues/2176

Should I open a distinct issue for the 1st bug, regarding `filters.py` & `color_space`?

**Comment by MartinThoma** [comment]
I will not merge the change done with `__main__` as pypdf should not become an application. We have `pdfly` for that.

However, the change to `filters` and maybe a part of the test looks unrelated / might be interesting to merge. Do you want to (1) adjust the PR or (2) open a new PR or (3) should I take care of it?

**Comment by MartinThoma** [comment]
@Lucas-C How should we continue with this PR?

**Comment by MartinThoma** [comment]
@Lucas-C Friendly reminder :-) How do you want to continue with this PR?

**Comment by MartinThoma** [comment]
I'm closing this PR as I assume that this will not have a follow-up. If I'm wrong, we can still re-open :-)
